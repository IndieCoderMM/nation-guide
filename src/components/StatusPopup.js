import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const StatusPopup = ({ title, message, showHome }) => (
  <div className="status-popup">
    <h2>{title}</h2>
    <p>{message}</p>
    {showHome && (
      <Link to="/" className="home-btn">
        Back to Home
      </Link>
    )}
  </div>
);

StatusPopup.propTypes = {
  title: PropTypes.string,
  message: PropTypes.string,
  showHome: PropTypes.bool,
};

StatusPopup.defaultProps = {
  title: '',
  message: '',
  showHome: false,
};

export default StatusPopup;
