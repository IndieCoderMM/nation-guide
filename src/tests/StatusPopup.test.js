import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import StatusPopup from '../components/StatusPopup';

describe('Popup Window', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<StatusPopup />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should display title and message', async () => {
    // Arrange
    const title = 'Hello';
    const message = 'Welcome to my app!';
    // Act
    render(<StatusPopup title={title} message={message} />);
    await screen.findByRole('heading');
    await screen.findByText(message);
    // Assert
    expect(screen.getByRole('heading')).toHaveTextContent(title);
    expect(screen.getByText(message)).toBeInTheDocument();
  });
});
