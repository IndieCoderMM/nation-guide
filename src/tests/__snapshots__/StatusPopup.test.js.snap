// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup Window should render correctly 1`] = `
<DocumentFragment>
  <div
    class="status-popup"
  >
    <h2 />
    <p />
  </div>
</DocumentFragment>
`;
